serpentTools/parsers/results.py

```diff
@@ -1,15 +1,319 @@
+
+    def _preProcessData(self):
+        """Pre-process data before collecting it"""
+        self._versionSerpent()  # Get the utilized Serpent version
+        self._numUnivSteps()  # Count the number of universes repetitions for memory allocation
+        vals = [self._univID[item] for item in self._univID]
+        if vals == {}:
+            raise SerpentToolsException("No universes are present in the file.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        if len(vals) > 1 and not vals[1:] == vals[:-1]:
+            raise SerpentToolsException("The repetitions of universes in the file is not equal.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        self._ChunkPosition()  # identify the 1st block for defining the dictionary
+        if self._ChunkPos == 0:
+            raise SerpentToolsException("The results file is corrupted.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        self._defineResultDict()  # Obtain the names of the variables to be stored
+        if self._resKeys == []:
+            raise SerpentToolsException("No variables appear in the results file.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+
+    def _defineResultDict(self):
+        """ Initializes the dictionary for ALL results included in .res.m"""
+        self._idxSect = [False, False, False]
+        self._resKeys = []  # Includes a complete set of .res variables
+        fObj = open(self.filePath, 'r')
+        numChar = 0  # count the number of characters
+        nUniv = len(self._univID)  # number of steps
+        nSteps = [self._univID[item] for item in self._univID][0]
+        for tline in fObj:
+            self._positionInFile(tline)
+            if numChar < self._ChunkPos[0]:
+                numChar += len(tline)
+                continue
+            # Not an empty line and not a comment
+            if tline.strip() != '' and not re.search(r'\%', tline):
+                currKey = re.search(r'^\w+', tline)
+                if currKey:
+                    keyName = convertVariableName(currKey.group())
+                    self._initResData(keyName, nSteps, tline)
+            numChar += len(tline)
+            if numChar >= self._ChunkPos[1]:
+                break
+        fObj.close()
+        self._idxSect = [False, False, False]
+
+    def _initResData(self, key, nSteps, tline):
+        """Initializes the dimensions of the variable"""
+        if key not in self._resKeys:
+            # Type of data (general meta, universes)
+            self._resKeys.append(key)
+            nvals = len(self._getDictVal(tline))
+            if self._idxSect[0]:
+                self.metadata[key] = ''
+            elif self._idxSect[1]:
+                self.resdata[key] = \
+                    empty((nSteps, nvals))
+
+    def _versionSerpent(self):
```

Comment:
Can you change this to something like `_getSerpentVersion`? This gives better insight into the purpose of the method